website/docs/concepts/scopes.mdx

```diff
@@ -0,0 +1,331 @@
+---
+title: Scopes
+---
+
+import Tabs from "@theme/Tabs";
+import TabItem from "@theme/TabItem";
+
+Scoping in Riverpod is a very powerful feature, but like all powerful features, it should be used wisely and intentionally.
+
+Scoping allows you to create new instances of the same provider in different parts of the tree, just like `package:provider` does.
+
+However, before scoping every provider you use, consider carefully why you want to scope the provider.
+In order to do so, you first need to understand what scoping means.
+
+## ProviderScope and ProviderContainer
+
+A scope is introduced by a [ProviderContainer]. This container holds the current state of all of your providers. 
+It manages the lookup and subscriptions between providers.
+
+In Flutter you should use the [ProviderScope] widget, which contains a [ProviderContainer]
+internally, and provides a way to access that container to the rest of the widget tree.
+
+```dart
+final valueProvider = StateProvider((ref) => 0);
+
+// DO this
+void main() {
+  runApp(ProviderScope(child: MyApp()));
+}
+
+//DON'T do this:
+final myProviderContainer = ProviderContainer();
+void main(){
+  runApp(MyApp());
+}
+```
+
+:::warning
+Do not use multiple [ProviderContainer]s, without an understanding of how they work. 
+Each will have it's own separate thread of states, which will not be able to access each other.
+Tests are an example of when you might want to use separate [ProviderContainer]s
+in order to make each test's state independent of the others.
+:::
+
+Only create a [ProviderContainer] without a [ProviderScope] for testing and dart-only usage.
+
+## How Riverpod Finds a Provider
+
+When a widget or provider requests the value of a provider, riverpod looks up the state of that provider in the [ProviderContainer] in the nearest 
+[ProviderScope] widget. If neither the provider nor one of it's explicitly listed dependencies is overridden in that scope riverpod continues it's lookup up the widget tree. 
+If the provider has not been overridden in any Widget subtrees the lookup defaults to the [ProviderContainer] in the root [ProviderScope].
+
+Once this process locates the scope in which the provider should reside it determines if the provider has been created yet. 
+If so, it will return the state of the provider.
+However, if the provider has been invalidated or is not currently initialized it will create the state using the provider's build method.
+
+## Scoping Providers to a Widget Subtree
+
+All providers can be scoped to a widget subtree. However, there are some things to keep in mind while doing so. Especially when it comes to understanding how the dependencies property works.
+
+The easiest way to understand subtree scoping is to go through a todo list example with a simple scoped provider,
+and refactor it to use a scoped provider with dependencies.
+
+
+Let's start with a simple TODO data object and a list of TODOs 
+```dart
+/// A simple Todo class
```

Comment:
Snippets should all be extracting to dart files and imported, for the sake of running the analyzer on them.